components/proxy/src/test/scala/com/hotels/styx/routing/handlers/ConditionRouterConfigSpec.scala

```diff
@@ -58,24 +58,76 @@ class ConditionRouterConfigSpec extends FunSpec with ShouldMatchers with Mockito
       |          content: "insecure"
       |""".stripMargin)
 
+  val configWithReferences = configBlock(
+    """
+      |config:
+      |    name: main-router
+      |    type: ConditionRouter
+      |    config:
+      |      routes:
+      |        - condition: protocol() == "https"
+      |          destination: secureHandler
+      |      fallback: fallbackHandler
+      |""".stripMargin)
+
+
   it("Builds an instance with fallback handler") {
-    val router = new ConditionRouter.ConfigFactory().build(List(), objectFactory, config)
+    val router = new ConditionRouter.ConfigFactory().build(List(), routeHandlerFactory, config)
     val response = router.handle(httpsRequest, null)
       .toBlocking
       .first()
 
-    response.status() should be (OK)
+    response.status() should be(OK)
   }
 
   it("Builds condition router instance routes") {
-    val router = new ConditionRouter.ConfigFactory().build(List(), objectFactory, config)
+    val router = new ConditionRouter.ConfigFactory().build(List(), routeHandlerFactory, config)
     val response = router.handle(httpRequest, null)
       .toBlocking
       .first()
 
     response.status().code() should be(301)
   }
 
+  it("Fallback handler can be specified as a handler reference") {
+    val routeHandlerFactory = new RouteHandlerFactory(
+      Map[String, HttpHandlerFactory](),
+      Map(
+      "secureHandler" -> new HttpHandlerAdapter(_ => Observable.just(Builder.response(OK).header("source", "secure").build())),
```

Comment:
Please import methods `just` and `response` directly to save space instead of qualifying with classes.
This comment applies to other similar lines in this file too.